tests/compatibility_tests/compatibility_test_runner.py

```diff
@@ -39,56 +43,39 @@ def load_all_subclasses(package_name):
     """
+    global _imported_modules
     package = importlib.import_module(package_name)
     for _, name, is_pkg in pkgutil.walk_packages(package.__path__, f"{package_name}."):
-        logger.debug(f"Processing this {name}, {is_pkg}")
+        logger.debug(f"Processing subclass: {name}, is package? -> {is_pkg}")
         if is_pkg:
             continue
+        if name in _imported_modules:
+            logger.debug(f"Skipping already imported module: {name}")
+            continue
         try:
             importlib.import_module(name)
+            _imported_modules.add(name)
             logger.debug(f"Imported module with name {name}")
         except ImportError as e:
             logger.error(f"Could not import module with {name}")
 
-def run_tests_in_mode(mode: Mode, from_version: str):
-    """
-    This method will be used to either run tests in prepare mode (i.e run prepare method of the resp. test case)
-    or run tests in test mode (i.e run test methods of the resp. test case).
-    """
-    logger.info(f"Running tests in '{mode}' mode with from_version: {from_version}")
-
-    if mode == Mode.PREPARE:
-        run_prepare_mode(from_version)
-    elif mode == Mode.TEST:
-        run_test_mode(from_version)
-
-def run_full_test_suite(from_version: str, to_version: str):
-    logger.info(f"Running full test suite with from_version: {from_version}, to_version: {to_version}")
-    run_prepare_mode(from_version)
-    run_test_mode(from_version)
-    logger.info(
-        "Finished tests across versions as part of Rollback test's full tests suite. THIS MARKS THE END OF ROLLBACK TESTS ACROSS TWO CONTAINERS WITH DIFFERENT VERSIONS")
-    full_test_run(to_version)
-
-def full_test_run(marqo_version: str):
-    """
-    This method will run tests on a single Marqo version container, which means it will run both prepare and tests on the
-    to_version Marqo container. Note that to_version Marqo container has been created by transferring instance from a
-    previous from_version Marqo container.
-    """
-    logger.info(f"Running full_test_run with version: {marqo_version}. THIS WILL RUN PREPARE METHOD AND TEST METHODS ON THE SAME {marqo_version} VERSION CONTAINER")
-    #Step 1: Run tests in prepare mode
-    run_prepare_mode(marqo_version)
-    #Step 2: Run tests in test mode
-    run_test_mode(marqo_version)
-
 def run_prepare_mode(version_to_test_against: str):
-
+    logger.info(f"===================================== RUN PREPARE MODE BEGINS =================================================")
     version_to_test_against = semver.VersionInfo.parse(version_to_test_against)
-    load_all_subclasses("tests.compatibility_tests")
     logger.debug(f"Printing all test cases defined under tests/compatibility_tests/: {BaseCompatibilityTestCase.__subclasses__()}")
     errors = []
+
+    # Skip any tests that have already been prepared
+    seen_classes = set()
```

Comment:
why do we need to check it twice? does the logic with `_imported_modules` not suffice?